### PR TITLE
Clean up some string.Format usage

### PR DESCRIPTION
**Pull request by stephentoub**
Mainly changes some string.Format usage to be SR.Format when working with resource strings.  Also cleans up a few asserts that were using string.Format.

Contributes to https://github.com/dotnet/corefx/issues/35001
Related to https://github.com/dotnet/arcade/pull/2158
Related to https://github.com/dotnet/corefx/pull/35777

### CODE REVIEW

File: src/System.Private.CoreLib/shared/System/Globalization/GregorianCalendarHelper.cs [hunk 1]
```diff
@@ -182,7 +182,7 @@ private int GetYearOffset(int year, int era, bool throwOnError)
                     {
                         throw new ArgumentOutOfRangeException(
                                     nameof(year),
-                                    string.Format(
+                                    SR.Format(
                                         CultureInfo.CurrentCulture,
```
Comment (jkotas) [inline src/System.Private.CoreLib/shared/System/Globalization/GregorianCalendarHelper.cs]: Do we really need to pass in `CultureInfo.CurrentCulture` explicitly in all these places? I would expect that `string.Format` is going to use current culture by default.

Comment (stephentoub) [reply src/System.Private.CoreLib/shared/System/Globalization/GregorianCalendarHelper.cs]: Agreed.  I've gone through and removed those.  I also realized I neglected to look at src/, only shared/, so I've cleaned up that, too.

File: src/System.Private.CoreLib/src/System/Reflection/CustomAttribute.cs [hunk 2]
```diff
@@ -694,7 +694,7 @@ private static RuntimeType ResolveType(RuntimeModule scope, string typeName)
 
             if (type == null)
                 throw new InvalidOperationException(
-                    string.Format(CultureInfo.CurrentUICulture, SR.Arg_CATypeResolutionFailed, typeName));
+                    SR.Format(CultureInfo.CurrentUICulture, SR.Arg_CATypeResolutionFailed, typeName));
```
Comment (jkotas) [inline src/System.Private.CoreLib/src/System/Reflection/CustomAttribute.cs]: My guess is that `CurrentUICulture` is used by accident in these, and it should be fine to delete it and just use default as well.

Comment (jkotas) [reply src/System.Private.CoreLib/src/System/Reflection/CustomAttribute.cs]: The formatting of typeNames, Types and other reflection types is not culture specific, so the explicit culture should not be needed when these items are involved.

Comment (stephentoub) [reply src/System.Private.CoreLib/src/System/Reflection/CustomAttribute.cs]: I'll go through and remove the cultures in such places.

File: src/System.Private.CoreLib/src/System/RtType.cs [hunk 3]
```diff
@@ -1776,8 +1776,8 @@ internal static MethodBase GetMethodBase(RuntimeType reflectedType, RuntimeMetho
                     }
 
                     if (!loaderAssuredCompatible)
-                        throw new ArgumentException(string.Format(
-                            CultureInfo.CurrentCulture, SR.Argument_ResolveMethodHandle,
+                        throw new ArgumentException(SR.Format(
+                            SR.Argument_ResolveMethodHandle,
                             reflectedType.ToString(), declaredType.ToString()));
```
[the remark this answers is not in the document]
Comment (stephentoub) [reply src/System.Private.CoreLib/src/System/RtType.cs]: I'll remove them. (My phone just auto corrected "remove" to "leave"... talk about changing one's meaning :)

File: src/System.Private.CoreLib/shared/System/Type.Enum.cs [hunk 4]
```diff
@@ -32,7 +32,7 @@ public virtual bool IsEnumDefined(object value)
             if (valueType.IsEnum)
             {
                 if (!valueType.IsEquivalentTo(this))
-                    throw new ArgumentException(SR.Format(SR.Arg_EnumAndObjectMustBeSameType, valueType.ToString(), this.ToString()));
+                    throw new ArgumentException(SR.Format(SR.Arg_EnumAndObjectMustBeSameType, valueType, ToString()));
```
Comment (jkotas) [inline src/System.Private.CoreLib/shared/System/Type.Enum.cs]:
```suggestion
                    throw new ArgumentException(SR.Format(SR.Arg_EnumAndObjectMustBeSameType, valueType, this));
```

**Comment by stephentoub** [comment]
@dotnet-bot test Ubuntu x64 Checked CoreFX Test please

**Pull request merged**